### PR TITLE
[ML] Use the same chunking configurations for models in the Elasticsearch service

### PR DESCRIPTION
**Pull request by davidkyle**
The `multilingual-e5` family of models take prefix strings `query: ` and `passage: ` in their input, Elasticsearch automatically applies those strings at inference. The bug is that those prefix strings are appearing in the text chunks created for semantic text fields. 

ELSER and E5 chunk differently to the other inference services, because tokenisation is done in ES the chunking for these models counts tokens not chunks. The bug is in the chunking for E5 which correctly adds the prefix strings to each input text for inference but does not remove them from the returned text. It proved very complicated to extract the prefix strings so this PR changes to chunker to use the word based chunker used by the other services. This change has advantage of unifying the chunking process for all models used in semantic text.

### CODE REVIEW

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elasticsearch/ElasticsearchInternalService.java [hunk 1]
```diff
@@ -429,46 +428,57 @@ public void chunkedInfer(
...
+            } else {
+                chunkPartListener.onFailure(
+                    createInvalidChunkedResultException(MlTextEmbeddingResults.NAME, inferenceResult.getWriteableName())
+                );
+                return;
+            }
         }
 
-        return translated;
+        chunkPartListener.onResponse(new InferenceTextEmbeddingFloatResults(translated));
```
[the remark this answers is not in the document]
Comment (davidkyle) [reply x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elasticsearch/ElasticsearchInternalService.java:475]: ML node models only produce float embeddings there is not support for byte embeddings yet. The sparse results are covered in the other branch of the conditional based on checking the task type.

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/chunking/EmbeddingRequestChunker.java [hunk 2]
```diff
@@ -240,6 +248,29 @@ private void handleByteResults(InferenceServiceResults inferenceServiceResults)
...
+                }
+
+                int start = 0;
+                for (var pos : positions) {
+                    sparseResults.get(pos.inputIndex())
+                        .setOnce(pos.chunkIndex(), sparseEmbeddings.embeddings().subList(start, start + pos.embeddingCount()));
+                    start += pos.embeddingCount();
+                }
+
+                if (resultCount.incrementAndGet() == totalNumberOfRequests) {
```
[the remark this answers is not in the document]
Comment (davidkyle) [reply x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/chunking/EmbeddingRequestChunker.java:290]: The code relies on the fact that for every request made the action listener will be called. Errors and the action timing out will call the onFailure method of the action listener. Working with that assumption then if `totalNumberOfRequests` are made then there will be `totalNumberOfRequests`, even if one or more of the requests time out.

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elasticsearch/ElasticsearchInternalService.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (dan-rubinstein) [reply x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elasticsearch/ElasticsearchInternalService.java]: Makes sense thanks for clarifying.

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/chunking/EmbeddingRequestChunker.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (dan-rubinstein) [reply x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/chunking/EmbeddingRequestChunker.java:290]: Makes sense. Thanks for clarifying.

**Review by dan-rubinstein — APPROVED** [review]

**Pull request merged**